src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1203,10 +1203,23 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                             }
 
                             var factory = new SyntheticBoundNodeFactory(methodSymbol, syntax, compilationState, diagsForCurrentMethod);
-                            var boundStatementsWithNullCheck = LocalRewriter.ConstructNullCheckedStatementList(methodSymbol.Parameters, boundStatements, factory);
-                            if (!boundStatementsWithNullCheck.IsDefault)
+
+                            // Iterators handled in IteratorRewriter.cs
+                            if (!methodSymbol.IsIterator)
                             {
-                                boundStatements = boundStatementsWithNullCheck;
+                                var boundStatementsWithNullCheck = LocalRewriter.ConstructNullCheckedStatementList(methodSymbol.Parameters, boundStatements, factory);
+
+                                if (!boundStatementsWithNullCheck.IsDefault)
+                                {
+                                    boundStatements = boundStatementsWithNullCheck;
+                                }
+                                hasErrors = boundStatementsWithNullCheck.HasErrors() || diagsForCurrentMethod.HasAnyErrors();
+                                SetGlobalErrorIfTrue(hasErrors);
+                                if (hasErrors)
+                                {
+                                    _diagnostics.AddRange(diagsForCurrentMethod);
+                                    return;
+                                }
```

Comment:
Can this be moved inside `if` above? (It seems like we only need to execute this if `ConstructNullCheckedStatementList` made a change.)